client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/index.jsx

```diff
@@ -70,9 +73,26 @@ class VisibleSubmissionEditIndex extends Component {
 
     return (
       assessment.timeLimit &&
-      submission.workflowState === 'attempting' && (
+      submission.workflowState === 'attempting' &&
+      (submission.timerStartedAt ? (
         <TimeLimitBanner submissionTimeLimitAt={submissionTimeLimitAt} />
-      )
+      ) : (
+        <Banner
+          className="bg-red-700 text-white border-only-b-fuchsia-200 fixed top-0 right-0"
+          icon={<HourglassTop />}
+        >
+          <FormattedMessage
+            {...translations.remainingTime}
+            values={{
+              timeLimit: (
+                <RemainingTimeTranslations
+                  remainingTime={assessment.timeLimit * 60 * 1000}
+                />
+              ),
+            }}
+          />
+        </Banner>
+      ))
```

Comment:
`TimeLimitBanner` already has some logic inside it, shouldn't you also include the additional `.timerStartedAt` check inside `TimeLimitBanner`, instead of bringing the logic out here in the `SubmissionEditIndex` page?

Reply:
done.. at first I don't really keen to couple the logic of not-yet-starting timer with when the timer already started. But I found a way to elegantly implement this component so that it includes both cases while keeping the code clean inside